src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -245,64 +252,82 @@ private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<
             }
             issuerHandles[_privateIntermediateCertificates.Length] = _rootCertificate.Handle;
 
-            using (SafeOcspRequestHandle ocspRequest = Interop.Crypto.X509BuildOcspRequest(subject, issuer))
-            {
-                byte[] rentedBytes = ArrayPool<byte>.Shared.Rent(Interop.Crypto.GetOcspRequestDerSize(ocspRequest));
-                int encodingSize = Interop.Crypto.EncodeOcspRequest(ocspRequest, rentedBytes);
-                ArraySegment<byte> encoded = new ArraySegment<byte>(rentedBytes, 0, encodingSize);
+            TaskCompletionSource<byte[]?> completionSource = new TaskCompletionSource<byte[]?>();
 
-                ArraySegment<char> rentedChars = UrlBase64Encoding.RentEncode(encoded);
-                byte[]? ret = null;
+            _pendingDownload = completionSource.Task;
+            FetchOcspAsyncCore(completionSource);
+            return completionSource.Task;
```

Comment:
Does anything bad happen if something starts waiting on `_pendingDownload` while the TCS's task is still PendingActivation?